source/utils/date.cpp

```diff
@@ -105,8 +105,16 @@ int date::to_number(calendar base_date) const
 
 date date::today()
 {
-    std::tm now = detail::localtime_safe(std::time(nullptr));
-    return date(1900 + now.tm_year, now.tm_mon + 1, now.tm_mday);
+    optional<std::tm> now = detail::localtime_safe(std::time(nullptr));
+
+    if (now.is_set())
+    {
+        return date(1900 + now.get().tm_year, now.get().tm_mon + 1, now.get().tm_mday);
+    }
+    else
+    {
+        return date(1970, 1, 1);
```

Comment:
Is this the best "invalid" date? Maybe we should use a real invalid date like `date(0,0,0)`? We can even add a `is_valid` or `is_null` member to `xlnt::date` to make it clear.

Reply:
I didn't like this solution either, and I think I should really change it right away. The only reason why I hesitated is because setting proper time limits is, unfortunately, a mess - thus, setting a default date is also a mess. In my experience:
1. Using anything outside of the years [0; 9999] causes crashes under MSVC depending on what you're doing (e.g. formatting dates using `std::format`). 
2. Using anything outside of the years [1970; 3000] (or [1970; 2038] under 32-bit) causes issues under MSVC.
3. Other platforms have other limits currently unknown to me (I could try it in the Compiler Explorer / Godbolt, though).

Any such solution would lead to some magic number (or magic date, in this case), which is definitely not nice either.

I think the only proper way to do it would be to have a function like `is_valid`. For example, we could allow default-constructing an `xlnt::date` object that would be invalid by default.

Reply:
Ok, for adding default constructor for invalid date. The only reason why I would prefer `is_null` instead of `is_valid` is that date(2000, 13, 35) is invalid too, but a simple `is_valid` check (e.g. year == date == month == 0) wouldn't say so.

Reply:
My newest commit contains your proposed changes. Let me know if you have any other suggestions!